src/element-style-observer.js

```diff
@@ -78,11 +85,18 @@ export default class ElementStyleObserver {
 		this.constructor.all.add(target, this);
 		this.properties = new Map();
 		this.target = target;
+		this.rootNode = new WeakRef(target.getRootNode());
 		this.callback = callback;
 		this.options = { properties: [], ...options };
 		let properties = toArray(options.properties);
 
 		this.renderedObserver = new RenderedObserver(records => {
+			if (this.target.getRootNode() !== this.rootNode.deref()) {
+				// The element was moved to another document or shadow root
+				this.rootNode = new WeakRef(this.target.getRootNode());
+				this.#setTreeCSS(this.rootNode.deref());
```

Comment:
We shouldn't need to dereference this again, setting the `rootNode` should suffice.

Reply:
In fact, this whole check should probably be in `#setTreeCSS()`.